tests/tests_unit/test_cdf_tk/test_auth.py

```diff
@@ -88,88 +82,78 @@ def mock_get_json(*args, **kwargs):
-    capfd,
+    cdf_resources: dict[type[CogniteResource], CogniteResource | CogniteResourceList],
 ):
-    # First add the pre-loaded data to the approval_client
+    # First, add the pre-loaded data to the approval_client
     for resource, data in cdf_resources.items():
         auth_cognite_approval_client.append(resource, data)
     # Then make sure that the CogniteClient used is the one mocked by
     # the approval_client
     cdf_tool_config.client = auth_cognite_approval_client.mock_client
-    check_auth(cdf_tool_config, group_file=Path(AUTH_DATA / "rw-group.yaml"))
-    out, _ = capfd.readouterr()
-    # Strip trailing spaces
-    out = "\n".join([line.rstrip() for line in out.splitlines()])
-    file_regression.check(out, encoding="utf-8", fullpath=to_fullpath(f"{TEST_PREFIX}_auth_verify_happypath"))
+    cmd = AuthCommand(print_warning=False)
+
+    cmd.check_auth(cdf_tool_config, group_file=Path(AUTH_DATA / "rw-group.yaml"))
 
-    dump = auth_cognite_approval_client.dump()
-    assert dump == {}
+    assert list(cmd.warning_list) == []
 
 
 def test_auth_verify_wrong_capabilities(
-    file_regression,
     cdf_tool_config: CDFToolConfig,
     auth_cognite_approval_client: ApprovalCogniteClient,
-    cdf_resources: dict[CogniteResource, Union[CogniteResource, CogniteResourceList]],
-    capfd,
+    cdf_resources: dict[type[CogniteResource], CogniteResource | CogniteResourceList],
 ):
+    expected_warnings = WarningList[ToolkitWarning]()
     # Remove the last 3 capabilities from the inspect result to make a test case
     # with wrong capabilities in the current CDF group.
     for _ in range(1, 4):
-        del cdf_resources[TokenInspection].capabilities[-1]
+        capability = cdf_resources[TokenInspection].capabilities.pop()
+        for cap in capability.capability.as_tuples():
+            expected_warnings.append(MissingCapabilityWarning(str(Capability.from_tuple(cap))))
     # Add the pre-loaded data to the approval_client
     for resource, data in cdf_resources.items():
         auth_cognite_approval_client.append(resource, data)
     # Then make sure that the CogniteClient used is the one mocked by
     # the approval_client
     cdf_tool_config.client = auth_cognite_approval_client.mock_client
-    check_auth(cdf_tool_config, group_file=Path(AUTH_DATA / "rw-group.yaml"))
-    out, _ = capfd.readouterr()
-    # Strip trailing spaces
-    out = "\n".join([line.rstrip() for line in out.splitlines()])
-    file_regression.check(out, encoding="utf-8", fullpath=to_fullpath(f"{TEST_PREFIX}_auth_verify_wrong_capabilities"))
+    cmd = AuthCommand(print_warning=False)
+
+    cmd.check_auth(cdf_tool_config, group_file=Path(AUTH_DATA / "rw-group.yaml"))
 
-    dump = auth_cognite_approval_client.dump()
-    assert dump == {}
+    assert len(cmd.warning_list) == len(expected_warnings)
+    assert set(cmd.warning_list) == set(expected_warnings)
```

Comment:
The new tests are nicer as they instead of checking whether the output was the same as last time, it is actually correct based on the setup of the test :)